type_inference/type_retrieval_service.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python
+#
+# Copyright 2023 Logica Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+from type_inference.bad_schema_exception import BadSchemaException
+from type_inference.postgresql_type_retriever import PostgresqlTypeRetriever
```

Comment:
Please follow the pattern of import that is used elsewhere like:

```
if '.' not in __package__:
  from common import color
  from compiler.dialect_libraries import recursion_library
  from parser_py import parse
else:
  from ..common import color
  from ..compiler.dialect_libraries import recursion_library
  from ..parser_py import parse
```

I recall this was necessary for it to work both when ran from repo and when ran from `setup.py` context.

Reply:
Actually it's not used everywhere. For example, meta_types.py, meta_types_test.py